geopandas/geodataframe.py

```diff
@@ -1420,6 +1430,14 @@ def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwargs):
             else:
                 if self.crs is not None and result.crs is None:
                     result.set_crs(self.crs, inplace=True)
+        elif isinstance(result, Series):
```

Comment:
This is needed because the new constructor_sliced wasn't strict and tried to upcast everything, including this case. I wonder if we should give a warning about the crs issue or not.